docs/proposals/volume-provisioning.md

```diff
@@ -0,0 +1,310 @@
+types of volumes within a single cloud.
+
+## Constraints and Assumptions
+
+The proposed design should:
+
+1.  Build upon the label selection capabilities added in Kubernetes 1.3
+2.  Not interfere with existing semantics of manually-created volumes
+3.  Not require users to know or understand the differences between
+    volumes (ie, Kubernetes should not dictate any particular set of
+    characteristics to administrators to think in terms of)
+
+This proposal will **not** deal with out-of-tree provisioners, but we should
+understand that they are desired by the community, and so we should be careful
+not to make them impossible to implement in the future.
+
+## Use Cases
+
+1.  As an administrator, I want to be able to control how volumes 
+
+### Use Case: Define how storage classes are provisioned
+
+Currently, there is an alpha version of dynamically provisioning, which allows
+provisioning of a **certain, single** volume type in clouds which offer
+natively supported persistent volumes.  For example, in clusters running on
+AWS, it is possible to provision AWS EBS volumes dynamically.  Real clusters
+require additional flexibility, such as the ability to differentiate different
+classes of storage to be provisioned, and the ability to provision volumes of
+different types.
+
+Using the new label selection feature for persistent volumes will allow
+storage classes to function as an extension of the existing API.  Storage
+classes should be matched with persistent volume claims using the same
+selector field as volumes, if no volumes are available that match a claim's
+selector.
+
+#### Multiple volume types
+
+Many administrators want the ability to dynamically provision multiple volume
+types (meaning, volumes that use different types of plugin) in the same
+cluster.  Some examples:
+
+1.  An administrator hosts their cluster in AWS and wants to provision EBS
+    volumes and GlusterFS volumes using a Gluster Virtual Storage Appliance
+    running in AWS
+2.  An administrator hosts their cluster in OpenStack and wants to provision
+    Cinder and NFS volumes
+
+In order to do this, the identity of the provisioner must be a property of the
+storage class (as opposed to implicit, as it is in the alpha feature).
+
+#### Parameterizing provisioners
+
+It will be common for a cluster to have multiple storage classes which
+leverage the same provisioner to create volumes with different characteristics
+for the different classes.  For example, an administrator running a cluster
+hosted in AWS might want to have two storage classes, `slow` and `fast`, that
+both leverage the same EBS provisioner to provision spinning disk volumes and
+provisioned-iops volumes, respectively.  From this, it follows that:
+
+1.  Storage classes should be able to hold a set of provisioner parameters
+2.  The provisioner should be able to get the details of the storage class
+    it is provisioning for
```

Comment:
`should be able to get` sounds like you are saying that the provisioner `MUST` retrieve the `storage class` rather than be passed the pertinent information. That seems like an implementation detail rather than the requirement for parmeterization. (not saying it is the wrong implementation detail per se.)